front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -30,25 +50,143 @@ export default function ImportTrainScheduleConfig(props: ImportTrainScheduleConf
   const dispatch = useDispatch();
   const { openModal, closeModal } = useContext(ModalContext);
 
+  async function importTrackSections(opIds: number[]): Promise<Record<number, TrackLocation[]>> {
+    const uniqueOpIds = Array.from(new Set(opIds));
+    const constraint = uniqueOpIds.reduce((res, uic) => [...res, ['=', ['uic'], Number(uic)]], [
+      'or',
+    ] as (string | SearchConstraintType)[]);
+    const payload: PostSearchApiArg = {
+      body: {
+        object: 'operationalpoint',
+        query: ['and', constraint, ['=', ['infra_id'], infraId]],
+      },
+      pageSize: 1000,
+    };
+    const operationalPoints = (await postSearch(
+      payload
+    ).unwrap()) as SearchOperationalPointResult[];
+
+    return operationalPoints.reduce(
+      (res, operationalPoint) =>
+        operationalPoint.uic
+          ? {
+              ...res,
+              [operationalPoint.uic]: operationalPoint.track_sections,
+            }
+          : res,
+      {}
+    );
+  }
+
+  function validateImportedTrainSchedules(
+    importedTrainSchedules: Record<string, unknown>[]
+  ): ImportedTrainSchedule[] | null {
+    const isInvalidTrainSchedules = some(importedTrainSchedules, (trainSchedule) => {
+      if (
+        !trainSchedule.trainNumber ||
+        !trainSchedule.rollingStock ||
+        !trainSchedule.departureTime ||
+        !trainSchedule.arrivalTime ||
+        !trainSchedule.departure ||
+        !trainSchedule.steps ||
+        !Array.isArray(trainSchedule.steps)
+      ) {
+        return true;
+      }
+      const hasInvalidateSteps = some(
```

Comment:
No need for lodash
`trainSchedule.steps.some(step => …)`

Reply:
Done !